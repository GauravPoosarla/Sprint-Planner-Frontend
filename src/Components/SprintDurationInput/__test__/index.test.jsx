import SprintDurationInput from '..';
import { fireEvent, getByTestId, render, screen } from '@testing-library/react';
import React from 'react';

describe('SprintDurationInput', () => {
  it('should render without errors', () => {
    const { asFragment } = render(<SprintDurationInput setValue={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render with a value', () => {
    const { asFragment } = render(
      <SprintDurationInput value="2" setValue={() => {}} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should call setValue when the input changes', () => {
    const setValue = jest.fn();
    const { getByTestId } = render(
      <SprintDurationInput value={'2'} setValue={setValue} />,
    );
    const input = getByTestId('sprint-duration-input-field').children[0]
      .children[0];
    fireEvent.change(input, { target: { value: '3' } });
    expect(setValue).toHaveBeenCalledWith('3');
  });
});
