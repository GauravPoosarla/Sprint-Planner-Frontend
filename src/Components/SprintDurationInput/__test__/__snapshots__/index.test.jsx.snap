// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SprintDurationInput should render with a value 1`] = `
<DocumentFragment>
  <div
    class="sprint-duration-input"
    data-testid="sprint-duration-input"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
    >
      Sprint Duration(weeks): 
    </label>
    <form>
      <div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="sprint-duration-input-field"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="sprint-duration"
              placeholder="Enter Sprint Duration"
              type="number"
              value="2"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`SprintDurationInput should render without errors 1`] = `
<DocumentFragment>
  <div
    class="sprint-duration-input"
    data-testid="sprint-duration-input"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
    >
      Sprint Duration(weeks): 
    </label>
    <form>
      <div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="sprint-duration-input-field"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="sprint-duration"
              placeholder="Enter Sprint Duration"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
